lib/assets/javascripts/cartodb/models/tabledata.js

```diff
@@ -733,18 +743,34 @@ cdb.admin.CartoDBTableData = cdb.ui.common.TableData.extend({
     }
 
     // use get to fetch the schema, probably cached
-    this._sqlQuery(_.template('select * from (<%= sql %>) __wrapped limit 0')({ sql: sql }), function(data) {
-      // get schema
-      self.query_schema = self._schemaFromQueryFields(data.fields);
-      if (!self.table.isInSQLView()) {
-        if ('the_geom' in self.query_schema) {
-          delete self.query_schema['the_geom_webmercator'];
+    //TODO This is a hacky way to make sure the table data returns before the get schema call.
+    //we need the table schema_alias to before normal schema to inject the alias into the columns
+    var self = this;
+    setTimeout(function(){
```

Comment:
what do you mean with "is not related" I see how you use `schema_alias` to complete the fields information here: https://github.com/coyle5280/cartodb/blob/cb06d3b71c4efa5d51784e7648628c6cfabb859b/lib/assets/javascripts/cartodb/models/tabledata.js#L757

The thing is: why don't you raise this logic when schema_alias and when the sql schema is returned? when both ones have data you fill the alias.